src/main/java/org/jabref/gui/maintable/OpenExternalFileAction.java

```diff
@@ -24,32 +28,51 @@ public OpenExternalFileAction(DialogService dialogService, StateManager stateMan
         this.stateManager = stateManager;
         this.preferencesService = preferencesService;
 
-        this.executable.bind(ActionHelper.isFilePresentForSelectedEntry(stateManager, preferencesService)
-                                         .and(ActionHelper.needsEntriesSelected(1, stateManager)));
+        this.executable.bind(ActionHelper.hasLinkedFileForSelectedEntries(stateManager, preferencesService)
+                .and(ActionHelper.needsEntriesSelected(stateManager))
+        );
     }
 
+    /**
+     * Open all linked files of the selected entries. If opening too many files, pop out a dialog to ask the user if to continue.
+     * <br>
+     * If some selected entries have linked file and others do not, ignore the latter.
+     */
     @Override
     public void execute() {
         stateManager.getActiveDatabase().ifPresent(databaseContext -> {
             final List<BibEntry> selectedEntries = stateManager.getSelectedEntries();
 
-            if (selectedEntries.size() != 1) {
-                dialogService.notify(Localization.lang("This operation requires exactly one item to be selected."));
-                return;
+            List<LinkedFileViewModel> linkedFileViewModelList = new LinkedList<>();
+            LinkedFileViewModel linkedFileViewModel;
+
+            for (BibEntry entry:selectedEntries) {
+                for (LinkedFile linkedFile:entry.getFiles()) {
+                    linkedFileViewModel = new LinkedFileViewModel(
+                            linkedFile,
+                            entry,
+                            databaseContext,
+                            Globals.TASK_EXECUTOR,
+                            dialogService,
+                            preferencesService.getXmpPreferences(),
+                            preferencesService.getFilePreferences(),
+                            ExternalFileTypes.getInstance());
+
+                    linkedFileViewModelList.add(linkedFileViewModel);
+                }
+            }
+
+            // ask the user when detecting # of files > FILES_LIMIT
+            if (linkedFileViewModelList.size() > FILES_LIMIT) {
+                boolean continu = dialogService.showConfirmationDialogAndWait(Localization.lang("Opening large number of files"),
```

Comment:
rename variable to `continue`

Reply:
`continue` is a keyword of Java. I will change to another meaningful name.